rpgJavaInterpreter-core/src/test/resources/smeup/MUDRNRAPU00214.rpgle

```diff
@@ -0,0 +1,13 @@
+     V* ==============================================================
+     D* 07/06/24  nn.mm author
+     D* D spec defined LIKE PARM is not resolved when:
+     D* DEFINE statement throws an error: Error while creating data definition
+     V* ==============================================================
+     D MSG             S             10
+
+     D OLDIPG          S                   LIKE(§CRIPG)
+     C     OLDIPG        DSPLY
+     C     DTAARA        DEFINE    *LDA          £UDLDA
+     C*                   IN        £UDLDA
```

Comment:
Why is `IN` stat commented?

Reply:
It's commented because the `IN` statement is not implemented yet, causing `gradlew check` to fail prematurely. As it is not necessary to replicate the issue I commented it. We can also consider removing it from the test or adding a mock implementation of the `IN` statement.

Reply:
**gradlew check failure**
You could bypass this error by adding your program name in `testing_utils.kt` `(whiteListRegexp)` 

**it is not necessary to replicate**
It is just the missing implementation of `IN` that causes the failure in the resolution of `OLDIPG`.    
If I comment `IN`:
![image](https://github.com/smeup/jariko/assets/40103274/5e9ddbeb-f940-4460-8936-6decb33d3ed6)

If I don't comment `IN`:
![image](https://github.com/smeup/jariko/assets/40103274/facc9059-c021-4e53-8cd7-67fcfd5a393f)

that is the problem to fix.

Reply:
I have updated with the proper implementation of the test case as suggested. For some reason I was sure that the problem was still present without the `IN` but as you correctly pointed out it is not the case. In any case, I have also extended the implementation of `executePgmCallBackTest` to look only for the errors specified (this was needed to prove we don't suffer from the data reference issue anymore). Let me know if further improvements are needed.